.github/workflows/vmtest.yml

```diff
@@ -0,0 +1,54 @@
+name: VM Test
```

Comment:
I don't know that vmtest is the most accurate name for this (especially if we move the lint job here). How about we keep this main logic in `ci.yml`, then have a new `pull_request.yml` that does the call.